ffi/capi/src/provider.rs

```diff
@@ -297,7 +297,7 @@ macro_rules! call_constructor_unstable {
             ))?,
             $crate::provider::ICU4XDataProviderInner::Empty => $unstable(&icu_provider_adapters::empty::EmptyDataProvider::new(), $($args,)*),
             #[cfg(feature = "buffer_provider")]
-            $crate::provider::ICU4XDataProviderInner::Buffer(buffer_provider) => $unstable(&icu_provider::AsDeserializingBufferProvider::as_deserializing(buffer_provider), $($args,)*),
+            $crate::provider::ICU4XDataProviderInner::Buffer(buffer_provider) => $unstable(&icu_provider::serde::AsDeserializingBufferProvider::as_deserializing(buffer_provider), $($args,)*),
```

Comment:
Thought/Suggestion: merge `serde` and `buf`

Reply:
Happy to, I thought that as well.